tools/bcdb_mock/pkg/escrow/types/escrow.go

```diff
@@ -0,0 +1,137 @@
+		Released bool `bson:"released"`
+		// Canceled indicates that the reservation was canceled, either by the
+		// user, or because a workload deployment failed, which resulted in the
+		// entire reservation being canceled. As a result, an attempt was made
+		// to refund the client. It is possible for this to have failed.
+		Canceled bool `bson:"canceled"`
+	}
+
+	// EscrowDetail hold the details of an escrow address
+	EscrowDetail struct {
+		FarmerID      schema.ID `bson:"farmer_id" json:"farmer_id"`
+		TotalAmount   xdr.Int64 `bson:"total_amount" json:"total_amount"`
+		EscrowAddress string    `bson:"escrow_address" json:"escrow_address"`
+	}
+)
+
+// ReservationPaymentInfoCreate creates the reservation payment information
+func ReservationPaymentInfoCreate(ctx context.Context, db *mongo.Database, reservationPaymentInfo ReservationPaymentInformation) error {
+	col := db.Collection(EscrowCollection)
+	_, err := col.InsertOne(ctx, reservationPaymentInfo)
+	if err != nil {
+		if merr, ok := err.(mongo.WriteException); ok {
+			errCode := merr.WriteErrors[0].Code
+			if errCode == 11000 {
+				return ErrEscrowExists
+			}
+		}
+		return err
+	}
+	return nil
+}
+
+// ReservationPaymentInfoUpdate update reservation payment info
+func ReservationPaymentInfoUpdate(ctx context.Context, db *mongo.Database, update ReservationPaymentInformation) error {
+	filter := bson.M{"_id": update.ReservationID}
+	// actually update the user with final data
+	if _, err := db.Collection(EscrowCollection).UpdateOne(ctx, filter, bson.M{"$set": update}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ReservationPaymentInfoGet a single reservation escrow info using its id
+func ReservationPaymentInfoGet(ctx context.Context, db *mongo.Database, id schema.ID) (ReservationPaymentInformation, error) {
+	col := db.Collection(EscrowCollection)
+	var rpi ReservationPaymentInformation
+	res := col.FindOne(ctx, bson.M{"_id": id})
+	if err := res.Err(); err != nil {
+		if err == mongo.ErrNoDocuments {
+			return rpi, ErrEscrowNotFound
+		}
+		return rpi, err
+	}
+	err := res.Decode(&rpi)
+	return rpi, err
+
+}
+
+// GetAllActiveReservationPaymentInfos get all active reservation payment information
+func GetAllActiveReservationPaymentInfos(ctx context.Context, db *mongo.Database) ([]ReservationPaymentInformation, error) {
+	filter := bson.M{"paid": false, "expiration": bson.M{"$gt": schema.Date{Time: time.Now()}}}
+	cursor, err := db.Collection(EscrowCollection).Find(ctx, filter)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get cursor over active payment infos")
+	}
+	paymentInfos := make([]ReservationPaymentInformation, 0)
+	err = cursor.All(ctx, &paymentInfos)
+	if err != nil {
+		err = errors.Wrap(err, "failed to decode active payment information")
+	}
+	return paymentInfos, err
+}
+
+// GetAllAddresses gets all in use addresses from the escrowcollections
+func GetAllAddresses(ctx context.Context, db *mongo.Database) ([]string, error) {
```

Comment:
This probably requires pagination otherwise this can blow up since it doesn't use any filtering. I assume this collection can has so many addresses